.github/workflows/python-package.yml

```diff
@@ -11,39 +11,58 @@ on:
 
 jobs:
   build:
-
-    #runs-on: ubuntu-latest
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         python-version: [3.8, 3.9, '3.10']
         os: [ubuntu-latest]
     steps:
     - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
       with:
-        submodules: true
+        python-version: ${{ matrix.python-version }}
+    - name: Build check
+      run: |
+        pip install build twine
+        python -m build .
+        python -m twine check dist/*
+  
+  codestyle:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [3.8, 3.9, '3.10']
+        os: [ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v3
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v3
       with:
         python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
+    - name: Install package
       run: |
-        python -m pip install --upgrade pip
-        python -m pip install wheel
-        python -m pip install flake8 pytest numpy cython
-        
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-        
-        # To generate version.py and compiled utils_c
-        python setup.py build
-        cp build/lib*/grizli/utils_c/*so grizli/utils_c/
-        
+        pip install flake8
     - name: Lint with flake8
       run: |
-        # stop the build if there are Python syntax errors or undefined names
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics --exclude "scripts* build* docs* dist* examples*"
-        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-        # flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics --exclude "scripts* eazy-photoz* astropy_helpers*"
+
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [3.8, 3.9, '3.10']
+        os: [ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install package
+      run: |
+        pip install -e .
+        pip install pytest
```

Comment:
The CI tests seem to be failing because at [this point](https://github.com/ivastar/grizli/pull/7/checks#step:5:58), `scipy` is not pulled in.

Maybe we should explicitly add in `scipy` to `setup.cfg/install_requires` (so that `pip install -e .` above pulls it in) or specify it here along with `pytest`?

Reply:
Right.  And there's a few more that needed to be added.  I think I've now (below) added what is needed to get the unit tests to run in CI, but I'm certain there are more unaccounted for dependencies as documented in #5.

Baby steps.  😅

Reply:
So I've fixed this in a new commit.  CI now passes.